move_group/src/default_capabilities/move_action_capability.cpp

```diff
@@ -94,6 +93,7 @@ void move_group::MoveGroupMoveAction::executeMoveCallback_PlanAndExecute(const m
 
   if (planning_scene::PlanningScene::isEmpty(goal->planning_options.planning_scene_diff))
   {
+    context_->planning_scene_monitor_->syncSceneUpdates();
```

Comment:
The `isEmpty` above is likely true, so this will block the callback most of the time.
I would propose to move the sync before the isEmpty check in order to avoid potential "works if `planning_scene_diff` is empty, but not if I add an object there" issues.
The following code requires the current state anyway as begin state.